packages/widgets/src/_inputs/widget-multi-text-input.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState } from "react";
+import { Combobox, Pill, PillsInput, useCombobox } from "@mantine/core";
+
+import type { CommonWidgetInputProps } from "./common";
+import { useWidgetInputTranslation } from "./common";
+import { useFormContext } from "./form";
+
+export const WidgetMultiTextInput = ({ property, kind, options }: CommonWidgetInputProps<"multiText">) => {
+  const t = useWidgetInputTranslation(kind, property);
+  const combobox = useCombobox({
+    onDropdownClose: () => combobox.resetSelectedOption(),
+    onDropdownOpen: () => combobox.updateSelectedOptionIndex("active"),
+  });
+
+  const [search, setSearch] = useState("");
+
+  const form = useFormContext();
+  const inputProps = form.getInputProps(`options.${property}`);
+  const values = inputProps.value as string[];
+  const onChange = inputProps.onChange as (values: string[]) => void;
+
+  const handleRemove = (optionIndex: number) => {
+    onChange(values.filter((_, index) => index !== optionIndex));
+  };
+
+  const isValidUrl = React.useMemo(() => {
+    if (!options.validate) {
+      return true;
+    }
+
+    const validationResult = options.validate.safeParse(search);
+    return validationResult.success;
+  }, [search]);
+
+  return (
+    <Combobox store={combobox}>
+      <Combobox.DropdownTarget>
+        <PillsInput
+          label={t("label")}
+          description={options.withDescription ? t("description") : undefined}
+          onClick={() => combobox.openDropdown()}
+          /* hide the error when nothing is being typed since "" is not valid but is not an explicit error */
+          error={!isValidUrl && search.length !== 0 ? "Provided URL is not valid" : undefined}
```

Comment:
Any better way of loading this translation @Meierschlumpf ?

Reply:
I would use the error from zod. We already translate parts of it